pkg/domain/infra/abi/containers.go

```diff
@@ -1058,11 +1058,23 @@ func (ic *ContainerEngine) ContainerMount(ctx context.Context, nameOrIDs []strin
 			os.Exit(ret)
 		}
 	}
-	ctrs, err := getContainersByContext(options.All, options.Latest, nameOrIDs, ic.Libpod)
+	reports := []*entities.ContainerMountReport{}
+	// Attempt to mount named containers directly from storage,
+	// this will fail and code will fall through to removing the container from libpod.`
+	names := []string{}
+	for _, ctr := range nameOrIDs {
+		report := entities.ContainerMountReport{Id: ctr}
+		if report.Path, report.Err = ic.Libpod.MountStorageContainer(ctr); report.Err != nil {
```

Comment:
I don't think this is safe - this feels like it should come *after* the Libpod mount logic?

Reply:
With the check to see if this is a libpod container, this should work fine now.